API/common/service_impl/knowledge.py

```diff
@@ -28,44 +28,87 @@ class MatchRule(int, Enum):
     if len(keyword.split()) > 1 and len(results) < top_num and match_rule == MatchRule.All:
-        or_results = get_search_results(build_or_search_statement(keyword), source_filter, top_num, scope.get_search_fields())
+        or_results = get_search_results(build_or_search_statement(keyword), source_filter, top_num, scope.get_search_fields(), SearchType.FullText)
         append_results(results, or_results)
         results = results[:top_num]
     return results
 
 
 def get_search_results(
         search_statement: str, source_filter: List[ScenarioSourceType],
-        top: int = 5, search_fields: Optional[List[str]] = None):
+        top: int = 5, search_fields: Optional[List[str]] = None, search_type=SearchType.Semantic):
     service_endpoint = os.environ["SCENARIO_SEARCH_SERVICE_ENDPOINT"]
     search_client = SearchClient(endpoint=service_endpoint,
                                  index_name=os.environ["SCENARIO_SEARCH_INDEX"],
                                  credential=AzureKeyCredential(os.environ["SCENARIO_SEARCH_SERVICE_SEARCH_KEY"]))
 
     filter = " or ".join([f"(source eq {src})" for src in source_filter])
-    results = search_client.search(
-        search_text=search_statement,
-        filter=filter,
-        include_total_count=True,
-        search_fields=search_fields,
-        highlight_fields=", ".join(search_fields) if search_fields else None,
-        top=top,
-        query_type='full')
+    if search_type == SearchType.Semantic:
+
+        results = search_client.search(
+            query_answer='extractive',
+            query_caption='extractive',
+            query_language='en-us',
+            semantic_configuration_name='semanctic-config',
+            search_text=search_statement,
+            filter=filter,
+            include_total_count=True,
+            search_fields=search_fields,
+            query_caption_highlight=True,
+            highlight_fields=", ".join(search_fields) if search_fields else None,
+            top=top,
+            query_type='semantic')
+    elif search_type == SearchType.FullText:
+        results = search_client.search(
+            search_text=search_statement,
+            filter=filter,
+            include_total_count=True,
+            search_fields=search_fields,
+            highlight_fields=", ".join(search_fields) if search_fields else None,
+            top=top,
+            query_type='full')
     results = list(results)
     for result in results:
         result.pop("rid")
-        result["score"] = result.pop("@search.score")
-        result["highlights"] = result.pop("@search.highlights")
+        if search_type == SearchType.Semantic:
+            result["score"] = result.pop("@search.reranker_score")
+            captions = result.pop("@search.captions")
+            if captions:        
+                result["highlights"] = {"scenario": captions[0].highlights}
```

Comment:
Out of curiosity, may I ask why we only take the highlights from the first item of `captions`?

Reply:
During testing, I found that the results of semantic search highlighting is a list. (The highlighted results of full-text search is a dictionary)
And there seems to be no case where the list length > 1, so I just took the first item.
To be on the safe side, add logic to process the entire list, separated by `;`